back/src/search/search.controller.ts

```diff
@@ -0,0 +1,108 @@
+import { BadRequestException, Body, Controller, DefaultValuePipe, Get, HttpCode, HttpStatus, InternalServerErrorException, NotFoundException, Param, ParseIntPipe, Post, Query, Req } from '@nestjs/common';
+import { ApiOperation, ApiParam, ApiTags } from '@nestjs/swagger';
+import { Song } from '@prisma/client';
+import { SongService } from 'src/song/song.service';
+import { SearchSongDto } from './dto/search-song.dto';
+import { SearchService } from './search.service';
+
+@ApiTags('search')
+@Controller('search')
+export class SearchController {
+	constructor(private readonly searchService: SearchService, private readonly songService: SongService) {}
+
+	@ApiOperation({
+		summary: 'Get a song details by song name',
+		description: 'Get a song details by song name',
+	})
+	@Get('song/:name')
+	async findByName(@Param('name') name: string): Promise<Song | null> {
+		const ret = await this.searchService.songByTitle({ name })
+		if (!ret) throw new NotFoundException;
+		return ret;
+	}
+
+	@ApiOperation({
+		summary: 'Get songs details by advanced filter',
+		description: 'Get songs details by advanced filter',
+	})
+	@Post('song/advanced')
+	@HttpCode(200) // change from '201 created' to '200 OK' http default response code
+	async findAdvanced(@Body() searchSongDto: SearchSongDto): Promise<Song[] | null> {
+		try {
+			const ret = await this.searchService.findAdvanced({
+				albumId: searchSongDto.album ? + searchSongDto.album : undefined,
```

Comment:
This syntax is a bit hard to read, the +. Just run prettier (I think we should add this in the CI)